### ISSUE TITLE
Raw prisma engine protocol query

### ISSUE THREAD

**Issue by MulverineX** [medz/dart-orm#281]
There should be a supported way to run queries directly to the prisma engine.

Once #328 is resolved I'm planning on using this library as a host for an internal prisma client in JavaScript, running in QuickJS via an upcoming library, [mercury](https://github.com/openwebf/mercury). Its a rather novel/obscure usecase, but I would appreciate not having to use internal methods of the library that may change.

**Comment by medz** [comment]
Yes, I'm working on a new engine, but the API is no longer consistent with the Prisma TS/JS client 👉 https://github.com/medz/prisma-dart/issues/277

The decision is a difficult one, as new client constructs are made in a way that is suitable for Dart or other languages that do not have Union types. Of course, it's not absolute, I have plans to support Union types in Dart (https://github.com/dart-lang/language/blob/main/working/union-types/nominative-union-types.md) Make it consistent with the Prisma TS/JS client API, which makes learning Prisma easier.

But progress is indeed a bit slow now. The new client API, although suitable for Dart, has completely deviated from Prisma AST and I need to do a lot of development work in the generator.

Additionally, I needed to write a new query engine for the Prisma engine so that it could be embedded in Flutter.

The old version only allowed the use of the Prisma binary engine in Dart Native.

I plan to allow Prisma to be used in any environment in the new version. My progress is about 30%. Please give me another month or so to implement it. I think I can initially launch Flutter Prisma and Dart native Prisma.

**Comment by MulverineX** [comment]
So is the plan not to bind the rust engine binary for all platforms?

**Comment by medz** [comment]
> So is the plan not to bind the rust engine binary for all platforms?

Dart native compilation does not support including other binaries, the plan for this is this:

- Dart Native: Download dynamic libraries on first run
- Flutter: Use Flutter’s native plug-in as a dynamic library
- Web: Built-in platform-independent Data Proxy engine

Except for the Web, you need to use Rust to rebuild the dynamic library version of the Prisma engine.

**Comment by medz** [comment]
There's another reason: I don't know how to start an HTTP service on the platform Flutter is running on by running a platform binary, which is how the Prisma binary engine works.

**Comment by MulverineX** [comment]
Ideally, direct queries to the engine would be made, rather than having to use the HTTP server

**Comment by medz** [comment]
Closing as superseded by the new Dart-native ORM merged in #485. This issue belongs to the retired Prisma-based client or its earlier roadmap. The new implementation uses Dart model declarations, independent drivers and single-engine Dart migrations, with no Prisma engine dependency.

The old implementation is no longer being developed. This closure does not claim that every requested feature or platform has been implemented. Please open a new issue against the new implementation with a current reproduction or focused proposal; see the repository README and capability documentation for the supported scope.

Thank you for the report and discussion.
